### PR TITLE
Add DiskWipe support to nvme using sanitize

### PR DESCRIPTION
**Pull request by mmlb**
### What does this PR do

Adds support to nvme for wiping the drive using sanitize ops. Both crypto erase and block erase are supported. The appropriate sanitize action is chosen based on detected capabilities.

### How can this change be tested by a PR reviewer?

Run on a machine with an nvme device.

### CODE REVIEW

**Comment by mmlb** [comment]
depends on #129

**Review by joelrebel — DISMISSED** [review]

File: examples/nvme-wipe/main.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	disk = flag.String("disk", "/dev/nvmeX", "nvme disk to wipe")
+	dur  = flag.String("timeout", (1 * time.Second).String(), "time to wait for command to complete")
+)
+
+func main() {
+	flag.Parse()
+
+	timeout, err := time.ParseDuration(*dur)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "failed to parse timeout duration:", err)
```
[the remark this answers is not in the document]
Comment (mmlb) [reply examples/nvme-wipe/main.go:23]: I'm not a fan of log output for examples and human run clis most of the time. I'm usually not interested in requiring a human to parse json or kv pairs if its going out to console.

**Comment by mmlb** [comment]
@joelrebel PTAL again. I added use of ApplyWatermarks to Sanitize.

**Comment by mmlb** [comment]
I rebased on top of latest main and then realized I should probably add ApplyWatermarks verification to Sanitize and that caught a bug with the use of rand.Int (see commit message: [`a448a15` (#136)](https://github.com/metal-toolbox/ironlib/pull/136/commits/a448a15655bb338abb00e8de4f0f20e3befb111c)). I also ended up cleaning up some error values returned and added some tests. Later I noticed I still had a panic in nvme.go and realized I never pushed the fixes.

Anyways long story long, I've got a bunch of "random" stuff in here too that I noticed/ran into while fixing up this PR and working on nvme format and nvme ns-reset support. I can split the random fixes/refactors/changes into a separate branch/PR if you want but I'd like to get it merged asap to avoid further confusing myself :D. Or we can just leave it in this PR. Whatever you want.

**Comment by mmlb** [comment]
depends on https://github.com/metal-toolbox/ironlib/pull/141

File: examples/nvme-wipe/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmlb) [reply examples/nvme-wipe/main.go]: Ended up going with logrus instead of fmt since WipeDisk is taking logrus itself.

**Review by ofaurax — DISMISSED** [review]
I approve as it seems urgent (see slack) but I haven't reviewed the code

**Comment by mmlb** [comment]
@ofaurax no need to blind-approve. I've rebased on top of main so now this PR is *much* smaller :D

File: utils/nvme.go [hunk 2]
```diff
@@ -259,6 +265,79 @@ func parseSanicap(sanicap uint) ([]*common.Capability, error) {
 	return caps, nil
 }
 
+//go:generate stringer -type SanitizeAction
```
Comment (joelrebel) [inline utils/nvme.go:268]: Should we update the Makefile to run the generate command before lint and test?

**Review by joelrebel — APPROVED** [review]
thanks for this, looks good to me!

**Pull request merged**